Changelog.md

```diff
@@ -7,6 +7,7 @@
 - Add a `quic.Config` option to request truncation of the connection ID from a server
 - Add a `quic.Config` option to configure the source address validation
 - Add a `quic.Config` option to configure the handshake timeout
+- Implement `net.Conn`-style deadlines for streams
```

Comment:
This PR doesn't actually change the stream interface, so these functions aren't visible publicly unless you type-assert. Is this intentional? If so, this should be documented someplace.